lib/src/multi_search_result.dart

```diff
@@ -0,0 +1,17 @@
+import 'searchable.dart';
+
+class MultiSearchResult {
+  final List<Searcheable> results;
+
+  const MultiSearchResult({
+    required this.results,
+  });
+
+  factory MultiSearchResult.fromMap(Map<String, Object?> json) {
+    final list = json['results'] as List<Object?>;
+    final parsed = list
+        .cast<Map<String, Object?>>()

```

Comment:
Does this work as well?
```suggestion
        .cast<Map<String, SearchResult?>>()
```

Reply:
nope, same reason explained above